src/Domain/Model/MembershipApplication.php

```diff
@@ -38,9 +40,8 @@ public function __construct( ?int $id, string $type, Applicant $applicant, Payme
 		$this->applicant = $applicant;
 		$this->payment = $payment;
 		$this->donationReceipt = $donationReceipt;
-		$paymentMethod = $payment->getPaymentMethod();
-		if ( $paymentMethod instanceof BookablePayment ) {
-			$this->confirmed = $paymentMethod->paymentCompleted();
+		if ( $this->payment instanceof BookablePayment ) {
+			$this->confirmed = $this->payment->getValuationDate() !== null;
```

Comment:
```suggestion
		if ( $payment instanceof BookablePayment ) {
			$this->confirmed = $payment->isBooked();
```

`isBooked` is implemented in bookable payments, but not part of `bookablePayment`, we should add that. Valuation date is just one of the methods we're using.

Keep in mind that payment is a deprecated property. We might keep it as a parameter to do the bookable check (although I'd prefer 2 additional constructor params, one for confirmed and one for the payment id)